packages/core/src/Services/socket-general.js

```diff
@@ -158,14 +158,18 @@ const BinarySocketGeneral = (() => {
                 if (!['reset_password', 'new_account_virtual'].includes(msg_type)) {
                     if (window.TrackJS) window.TrackJS.track('Custom InvalidToken error');
                 }
+                common_store.setShouldSkipRaiseError(true);
                 client_store.logout().then(() => {
+                    window.location.href = '/';
```

Comment:
```suggestion
                     common_store.routeTo(routes.trade);
```
With this approach we don't need an extra observable `should_skip_invalid_token_error`.
Just this line and the change in [`connected-apps.jsx`](https://github.com/binary-com/deriv-app/pull/2031/files#diff-156e4b743a3de6b4db5ee0feeaea98f587ea836810a38aac9bf29dfabbfa9761L45) will do the trick. 
Can you try this please? 🙏

Reply:
Will give it a try. We had a similar discussion with @reza-fs on how to not use `flags` here.

Reply:
Thanks @balakrishna-binary 
It worked :tada: